crypto/asn1/i2d_pr.c

```diff
@@ -30,35 +31,23 @@ int i2d_PrivateKey(const EVP_PKEY *a, unsigned char **pp)
         }
         return ret;
     }
-    if (a->keymgmt != NULL) {
+    if (evp_pkey_is_provided(a)) {
+        /*
+         * This is horribly much, but then again, |*pp| is unbounded, and
+         * we really can't know better.
+         */
+        size_t length = INT_MAX;
         /* The private key includes everything */
         int selection =
             OSSL_KEYMGMT_SELECT_ALL_PARAMETERS | OSSL_KEYMGMT_SELECT_KEYPAIR;
-        OSSL_ENCODER_CTX *ctx =
-            OSSL_ENCODER_CTX_new_by_EVP_PKEY(a, "DER", selection, NULL, NULL);
-        BIO *out = BIO_new(BIO_s_mem());
-        BUF_MEM *buf = NULL;
         int ret = -1;
+        OSSL_ENCODER_CTX *ctx;
 
-        if (ctx != NULL
-            && out != NULL
+        if ((ctx = OSSL_ENCODER_CTX_new_by_EVP_PKEY(a, "DER", selection,
+                                                    NULL, NULL)) != NULL
```

Comment:
libctx and propq are NULL here...
Do we need a variant og the function that passes these in? This is a creator so I suspect so.